rikai/experimental/pg/model.py

```diff
@@ -12,56 +11,77 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-from typing import Optional
+import json
+from typing import Any, Callable, Dict, Optional
 
 import torch
-
+import torchvision
+import torchvision.transforms as T
+from rikai.parquet.dataset import convert_tensor
+from rikai.pytorch.models.torch import ClassificationModelType, ObjectDetectionModelType
+from rikai.pytorch.models.feature_extractor import FeatureExtractorType
 from rikai.spark.sql.codegen.dummy import DummyModelSpec
 from rikai.spark.sql.codegen.fs import FileModelSpec
 from rikai.spark.sql.model import ModelType
 from rikai.types import Image
+from torchvision.models.feature_extraction import create_feature_extractor
+
+from .logging import info
+from .schema import parse_schema
 
 
 class PgModel:
+    """PostgreSQL Model"""
+
     def __init__(self, model_type: ModelType):
         self.model = model_type
+        self.transform: Optional[Callable] = self.model.transform()
+
+        self._is_vision_type = isinstance(
+            self.model,
+            (ClassificationModelType, ObjectDetectionModelType, FeatureExtractorType),
+        )
+
+    def schema(self) -> str:
+        return self.model.schema()
+
+    def args(self) -> str:
```

Comment:
don't quite understand what this does, but maybe it'll become clear later

Reply:
This was about building the arg list for the PG UDF.